crates/polars-core/src/chunked_array/ops/sort/arg_sort.rs

```diff
@@ -12,24 +12,104 @@ where
         options.multithreaded,
     );
 }
+// Compute the indexes after reversing a sorted array, maintaining
+// the order of equal elements, in linear time. Faster than sort_impl
+//  as we avoid allocating extra memory.
+pub(super) fn reverse_stable_no_nulls<I, J, T>(iters: I, len: usize) -> Vec<IdxSize>
+where
+    I: IntoIterator<Item = J>,
+    J: IntoIterator<Item = T>,
+    T: TotalOrd + Send + Sync,
+{
+    let mut current_start: IdxSize = 0;
+    let mut current_end: IdxSize = 1;
+    let mut flattened_iter = iters.into_iter().flatten();
```

Comment:
We need to use `iter` as `into_iter` boxes.

I also don't like flattened iterators. If we can write this with explicitly looping over the chunks we should prefer that.

Reply:
Have rewritten this, let me know if it looks fine now.